### PR TITLE
Fix InvalidPathException when match string includes * #778

### PR DESCRIPTION
**Pull request by lucastheisen**
This is an alternative to #779 which is a little safer in that it captures any invalid path (not just "*", and not just windows).  It also [fixed some java url wonkiness](https://stackoverflow.com/a/17870390/516433) that caused other unit tests to fail for illegal char in path.

### CODE REVIEW

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java [hunk 1]
```diff
@@ -455,7 +455,8 @@ public void testPullPrivateRepoWithoutAuth() throws Exception {
   public void testBuildImageIdWithBuildargs() throws Exception {
     requireDockerApiVersionAtLeast("1.21", "build args");
 
-    final String dockerDirectory = Resources.getResource("dockerDirectoryWithBuildargs").getPath();
```
Comment (davidxia) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: Were these tests failing on Windows machines?

**Comment by lucastheisen** [comment]
@davidxia , its not :
```
final String dockerDirectory = Resources.getResource("dockerDirectoryWithBuildargs").getPath();
```
that fails, its later when `dockerDirectory` is used that it throws an exception (see [this comment](https://github.com/spotify/docker-client/pull/779#issuecomment-306052712) for details).  Specifically, on windows, `dockerDirectory` would have a value like `/C:/...` instead of `C:/...` which throws an `InvalidPathException` when you try to use it as a `Path` (and `Path.startsWith(string)` converts the string to a `Path` object before comparing).

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java (hunk 1 above)
Comment (lucastheisen) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: See reply in [comment below](https://github.com/spotify/docker-client/pull/780#issuecomment-306270015) (i didn't notice this reply box :) )

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java [hunk 2]
```diff
@@ -455,7 +455,8 @@ public void testPullPrivateRepoWithoutAuth() throws Exception {
...
+    final String dockerDirectory = 
+        Paths.get(Resources.getResource("dockerDirectoryWithBuildargs").toURI()).toString();
```
Comment (mattnworb) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: this line is repeated many times but with different directory names - can you wrap it in a method and also have a comment in the method explaining why the URL -> URI -> String -> Path dance is needed?

Comment (lucastheisen) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: Yeah, i did that on purpose...  I find it easier to get my pull requests accepted (across many different projects) if i minimize my changes and adopt the approach of the project.  So i changed each line, but definitely agree that creating a helper method makes more sense.  Let me do so real quick.  Should have an update in a little while.

**Comment by lucastheisen** [comment]
Argh!  Cut/paste error.  My bad... new version should be pushed in a second...

**Comment by mattnworb** [comment]
thanks!

**Pull request merged**